content/en/blog/_posts/2025-04-21-kubernetes-network-evolution/index.md

```diff
@@ -0,0 +1,71 @@
+---
+
+Kubernetes is in continuous evolution, and as a distributed system, networking is an inseparable aspect, requiring to evolve to support the platform's ongoing development. What began with a relatively simple, flat network model has progressively evolved to accommodate increasingly complex and demanding workloads. Understanding this journey, from the foundational choice of the Container Network Interface (CNI) over Docker's Container Network Model (CNM) to the latest advancements in Dynamic Resource Allocation (DRA) and Declarative Networking, is crucial for anyone operating within the Kubernetes ecosystem. This post illustrates this path, diving into the key decisions and events that have shaped Kubernetes networking, providing insights into the rationale behind the current state and the direction of future development.
+
+## The Genesis: Choosing CNI Over CNM
+
+In the early days of Kubernetes, the question of how to manage container networks was a significant one. Two primary models emerged: the Container Network Interface (CNI) and Docker's Container Network Model (CNM). Ultimately, the Kubernetes community opted for CNI, a decision rooted in fundamental architectural and practical considerations, as explained in Tim Hockin’s blog [Why Kubernetes Doesn't use libnetwork](https://blog.kubernetes.io/2016/01/why-kubernetes-doesnt-use-libnetwork/).
+
+The core principle behind CNI was to provide a flexible and focused interface for network plugins. The emphasis was on a standardized mechanism that allowed various network providers to integrate seamlessly with Kubernetes by implementing a simple set of executables. Early discussions in 2015 within the [`kubernetes-sig-network` mailing list](https://groups.google.com/g/kubernetes-sig-network/c/5MWRPxsURUw) reveal a strong desire for stable base plugins that could be extended to meet diverse networking needs. The pluggable design of CNI allowed for the integration of various network plugins, supporting different network topologies and requirements within Kubernetes.
+
+## The Road to Declarative Networking: Key Milestones
+
+With CNI established as the foundation, the Kubernetes community began to work on more advanced networking requirements, particularly the need for a more declarative networking support. This journey was marked by several key initiatives and discussions:
+
+### The Network Plumbing Working Group (NPWG)
+
+The Network Plumbing Working Group (NPWG) was formed at Kubecon NA '17 in Austin to initially work towards implementing a common standard addressing how one might attach multiple networks to pods in Kubernetes. The work of the NPWG has culminated in a [standard Multi-Network Custom Resource Definition (CRD)](https://github.com/k8snetworkplumbingwg/multi-net-spec). The [Multus CNI](https://github.com/k8snetworkplumbingwg/multus-cni) is the reference implementation of this multi-network definition, demonstrating how pods can have multiple network interfaces.
```

Comment:
You talked about "the need for a more declarative networking support" above, but Multus/NAD is _very_ imperative, which is in fact part of what kept it out of tree.